### DISCUSSION TITLE
How to setup HammerDB for Mariadb Testing.

### DISCUSSION THREAD

**Discussion by Prakashansible** [TPC-Council/HammerDB#80] [Q&A] [answered]
Hello All,
I'm able to setup and test MySQL using HammerDB and now I'm planning to Test Mariadb but not able to set db set parameters like 'db set maria' - its saying use correct prefix..
Can any one of please let me know how to setup mariadb for HammerDB..

**Comment by sm-shaw** [comment, marked as answer]
Hi, 
Yes there are a couple of ways to do this.  Firstly and probably the easiest way is to use the HammerDB MySQL option to test MariaDB. MariaDB is MySQL compatible so HammerDB for MySQL works absolutely fine against MariaDB. The only difference is that the client libraries are different so you need HammerDB to load the MySQL 8.0 libmysqlclient.so.21 for HammerDB v3.3 and libmysqlclient.so.20 for version 3.2 and earlier. This one MySQL library file is all you need in the LD_LIBRARY_PATH for the HammerDB client to work and test MariaDB. 

If however you want to add MariaDB fully then I covered this in this series of blog posts
https://www.hammerdb.com/blog/uncategorized/how-to-add-your-database-to-hammerdb-pt1-opening-an-issue/
and the code is here:
https://github.com/sm-shaw/HammerDB/tree/54
(Note that this is not fully updated for v3.3)
Therefore the template is there to add MariaDB to HammerDB fully. 

In the medium term the plan is to add MariaDB because although you can test it with the MySQL option MariaDB has features that diverge from MySQL such as the column store for the TPC-H workload and PL/SQL compatibility for OLTP.  It is more practical to support these under a MariaDB only option rather than under MySQL.  Also as above MariaDB has separate libraries so it also makes sense building the mysql interface against this so testers of MariaDB do not have to grab the MySQL library as well. 

for now however using the MySQL option is the best approach. Please advise if this works for you.

**Comment by Prakashansible** [comment]
Same settings of MySQL worked fine for Mariadb as well..Thank you so much Shaw..!!
